features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/members/details/RoomMemberDetailsNode.kt

```diff
@@ -95,11 +79,12 @@ class RoomMemberDetailsNode @AssistedInject constructor(
         val state = presenter.present()
 
         LaunchedEffect(state.startDmActionState) {
-            if (state.startDmActionState is AsyncAction.Success) {
-                onStartDM(state.startDmActionState.data)
+            val result = state.startDmActionState
```

Comment:
Not a blocker, but why is the `result` variable created here?

Reply:
Else smart cast is not possible

<img width="1165" alt="image" src="https://github.com/element-hq/element-x-android/assets/3940906/f66fbe9f-afbc-4a99-83a7-9a6ba1c6d3d2">